### DISCUSSION TITLE
120

### DISCUSSION THREAD

**Discussion by giscus** [CODe5753/giscus#1] [Comments]
# 120

서론 2023년 상반기에 이력서 스터디를 진행했고 프론트엔드를 제외한 직군에서 스터디 리딩을 하게 되었습니다. 약 10명 가까운 인원이 참여했으며, 이력서 플랫폼(또는 툴) 결정 과정과 제 이력서가 어떻게 바뀌었는지를 자세하게 기술합니다. 단, 스터디 진행 방식에 대한 회고글은 아님을 알려드립니다. 이력서 최종 업데이트 이후 지원한 모든 회사에 서류 합격을 할 수 있었습니다. 이력서와 관련된 글 보기 2023.07.16 - [후기/경험] - 2023 상반기를 되돌아보자 2023.12.10 - [후기/경험] - 채용 한파에 카카오뱅크 최종 합격 후기 이력서에 노션을 추천하지 않는 이유 2022.12.08 - [후기/넥스터즈] - 넥스터즈 지원자 서류 검토 후기 넥스터즈 회장을 하면서 수백개의 이력서를 하나..

https://code1995.tistory.com/120

**Comment by dkswnkk** [comment]
댓글은 결국 giscus로 결국 처리하셨군요!! 
글 잘 봤습니다 승현님! 어쩌다보니 https://imksh.com/119 후기글까지 보게 되었는데 축하드립니다ㅎㅎ

**Reply by CODe5753** [reply]
주형님 안녕하세요!
좋은 정보를 제공해주셔서 `giscus`를 도입하게 되었습니다 ㅎㅎ
쑥쓰럽지만, 축하 너무 감사합니다 😊
좋은 아티클 많이 남겨주세요! 종종 구경가겠습니다 ㅎㅎ

**Comment by Dongmin-Sim** [comment]
글 너무 잘읽었습니다 변화된 버전들을 보면서 이력서를 작성하는데 있어 검토해볼만한 기준들을 얻어 갈 수 있었던거 같아 도움이 되었습니다👍

**Reply by CODe5753** [reply]
동민님~ 도움이 되셨다니 뿌듯하네요!
다음에도 도움이 될만한 아티클을 작성해볼테니 자주 방문해주세요!
감사합니다 😊

**Comment by joonfluence** [comment]
와우 이런 글 너무 사랑합니다..🧡 저도 이 내용 보고 얼른 이력서 고치러 갑니다~!!! 🏃‍♂️🏃‍♂️

**Reply by CODe5753** [reply]
@joonfluence 
이력서 수정에 도움이 되면 좋겠습니다~!
감사합니다 ☺️

**Comment by AnneMayor** [comment]
아 너무 좋은 글 공유 감사합니다. 덕분에 제 이력서 작성하는 데 큰 도움이 되었네요:)
막막했던 길에 한줄기 빛과 같은 글이라고 생각합니다!

**Reply by CODe5753** [reply]
@AnneMayor
도움이 되셨다니 다행입니다!
다음에도 도움이 될 만한 글을 써볼게요! 
감사해요 😊

**Comment by howyoujini** [comment]
소중한 후기글 잘 읽었습니다! 경험에 비추어서 어떤 변화가 있었는지 확실히 눈에 보여서 더욱 재밌게 읽었습니다.
저도 덕분에 이력서를 수정해봐야겠네요! 감사합니다 승현님 :)

**Reply by CODe5753** [reply]
@howyoujini 
멋진 이력서 만드시면 좋겠어요!!
감사합니다 😊

**Comment by ooo1324** [comment]
시행착오를 겪지 않았으면 하는 마음으로, 정성들인 이력서와 경험을 공유해주셔서 진심으로 감사드립니다.
이력서의 틀과 방향을 잡는것에 많은 도움이 되었습니다!

저도 누군가에게 도움이 되는 글이나 경험을 공유하고 싶다는 생각이 드네요!

**Reply by CODe5753** [reply]
@ooo1324
도움이 되었다는 점에서도 뿌듯하지만
이 글을 통해 공유하고자 하는 생각까지 얻으셨다하시니 너무나 기쁘네요!
앞으로의 활동 기대할게요 감사합니다 :)

**Comment by Denia-park** [comment]
덕분에 좋은 정보 잘 보고 갑니다.

채용 한파라 요즘 취업이 쉽지 않은데, 승현님 글 보고 힘내서 계속 수정하고 도전해보겠습니다!

감사합니다.

**Reply by CODe5753** [reply]
@Denia-park
감사합니다!!
깃헙을 잠깐 봤는데 정말 열심히 노력하고 계시는 것 같아 되려 열정 얻고 갑니다!
응원하겠습니다 🔥

**Reply by Denia-park** [reply]
@CODe5753

혹시 실례가 안 되신다면, `Java Spring` 백엔드 취준생이 사이드 프로젝트에 적용하면 좋을 만한 내용 좀 추천해 주실 수 있으실까요 ???
이력서에 이런 내용이 담겨있으면 좋을 것 같다 같은 것들로 이야기를 해주시면 정말 큰 도움이 될 것 같습니다. (Ex. `모니터링`, `Redis`, `Docker` 등등)

최대한 되는대로 이것저것 적용해 볼 생각으로 사이드 프로젝트를 진행 중이긴 한데, 하고는 있지만 `"이걸로 정말 괜찮은 걸까?"`, `"내가 잘 준비하고 있는 게 맞나?"` 같은 생각들이 자꾸 들어서 실례를 무릅쓰고 여쭤봅니다.🙇‍♂️🙇‍♂️

**Reply by CODe5753** [reply]
@Denia-park 

동문서답일 수 있는데 **모두 다 적용되어도 좋고 모두 다 없어도 상관없습니다.**

**적용하는 것도 중요하겠지만, `왜?` 적용했는가가 중요합니다.**
왜 적용하려고 하시는건가요? _(열거하신 이유가 궁금한데 고민해보시고 답글 달아주시면 좋겠습니다 😊)_
여러가지 이유가 있을 수 있을 것 같은데, 그 이유 안에서도 여러 질문이 생길 수 있어요.
아래의 질문에 하나씩 파다보면 `지금 나에게 필요한 스택이 뭔지`에 대한 해답을 도출할 수 있을 것 같아요.

- 응답 속도가 느리네
  - 응답 속도가 느린게 뭐지? 무슨 기준으로 느리다고 판단하는거지?
  - 응답 속도를 빠르게하려면 어떻게 할 수 있을까?
    - 지금 돌아가는 쿼리가 최선일까?
      - 인덱스를 탔을 때 더 빠른 작업인데, 혹시 지금 인덱스를 타지 않고 풀스캔으로 작동중인건 아닌가?
        - 그렇다면 쿼리를 개선해보자
    - 읽기 작업이 느린거니까 캐시를 달아보는건 어떨까?
      - 캐시엔 어떤 종류가있지?
        - 로컬에서도 캐시를 쓸 수 있고, 다들 사용하는 Redis라는 것도 있구나
          - 로컬과 아닌것은 무슨 차이지?
      - 무작정 캐시를 달면 빠른가?
        - 무조건 빠른거면 모든 시스템에 캐시가 다 달려있나?

또 다른 관점으로 "모두 다 없어도 상관없다"라고 말씀드렸는데요.
`꼭 그 기술이 필요한가?`에 대해서도 고민해보시면 좋겠습니다.
- 오버엔지니어링은 아닌가?
  - 오버엔지니어링이라고 판단하는 기준은 뭘까?
- 이 기술 스택을 적용하는 것에 러닝커브가 있진 않은가?
  - 그럼 데드라인이 있는 프로젝트에서는 문제가 되진 않을까?
- 이 기술을 사용하지 않으면 안되는건가?
  - 대체할 수 있는 기술이 있을까?

**사용하는 것이 중요한 게 아니라 `왜?`라는 물음을 항상 스스로에게 던져보셔요.**

---

다시 돌아와서 명쾌한 답변을 드리자면, 응답 속도 개선을 위해 Redis를 먼저 사용해보셔요.
고민할 부분이 많을거에요~!
화이팅입니다 ㅎㅎ

**Reply by Denia-park** [reply]
@CODe5753 

먼저 정성 가득한 답변 정말 감사합니다.

적용하려고 했던 이유는 사실 요즘 채용 공고 보면 제일 많이 나와있는 내용들이라서 적용을 해보려고 했던 이유가 가장 큽니다.😅

이력서용으로 혼자서 하는 사이드 프로젝트다보니 최대한 트러블 슈팅할 내용들이 많이 나오는게 필요하다고 생각했습니다.
그래서 해당 기술 스택의 필요성보다는 `자격요건이랑 우대사항에는 적혀있으니 이런 기술들을 해둬야 하지 않을까?` 라는 생각으로 적용하려고 한 이유가 제일 커서 왜 필요하지? 라는 물음에 대해서는 주의깊게 생각을 하지 않았던 것 같습니다.

이번 기회에 다시 한번 말씀해주신 내용들을 기반으로 잘 고민해보고 이력서에 녹여낼 수 있도록 한번 해보겠습니다!!!

다시 한번 귀한 시간 내주셔서 정말 감사합니다. 🙇‍♂️🙇‍♂️

**Reply by CODe5753** [reply]
전략적으로 잘 접근하고 계신 것 같습니다!
이상적인 답변을 드렸지만 취업은 현실이니까요 😢
다만 면접관이 납득하지 못하는 기술스택 사용은 질문으로 꼭 들어오더라구요.
그 때 답변을 잘하셔야 합니다. ("그냥 써봤다"도 좋지만 논리적인 이유가 있다면 더 좋겠죠!)

@Denia-park 님의 노력을 진심으로 응원합니다!
좋은 소식 들려주세요 😊

**Reply by Denia-park** [reply]
@CODe5753

안녕하세요. 승현님!
약 2달만에 다시 인사를 드립니다.

승현님의 이력서 작성 글 덕분에 이번에 서비스 회사에 취업 할 수 있었습니다.
승현님의 소중한 경험 및 지식을 이렇게 나눠주셔서 정말 감사합니다!! 🙇‍♂️🙇‍♂️

알려주신 내용처럼 이력서는 최대한 깔끔하게 작성하고,
제가 사용한 기술 및 트러블 슈팅에 대해서 `왜?` 했는지에 대해 고민 해보는게 정말 중요한 것 같습니다.

다시 한번 감사합니다!!
그리고 앞으로도 좋은 글 많이 부탁드립니다!!

**Reply by CODe5753** [reply]
@Denia-park 
안녕하세요!
우선 진심으로 너무 축하드립니다 🎉
이 글이 그저 스쳐 지나가는 글일 수 있었을텐데 이렇게 기쁜 소식까지 공유해주셔서 너무 감사드려요
덩달아 저도 기분이 좋군요 ㅎㅎ

이렇게 감사 표현을 하시는 모습에 저또한 감사드린다는 말씀 전하며
앞으로의 행보, 진심으로 응원하겠습니다. 😊

**Comment by tneserp** [comment]
SSAFY 중도퇴소 글을 통해 들어왔다가 좋은 글을 많이 접하게 되네요 :-) 

현재 취준생으로서, 신입 지원 할 때 포트폴리오 제출할 일이 많은데요. 현재 notion을 통해 포트폴리오를 관리하고 있는데, docx로 관리 하는게 좋을까요?

혹은 포트폴리오에 관해 해주실 조언이 있을까요?

**Reply by CODe5753** [reply]
@tneserp 님 안녕하세요!
좋은 말씀 감사합니다 :)

이력서와 별개로 개발자 포폴은 notion으로 관리해도 괜찮다는 생각이 듭니다!
제가 생각하는 이유는 아래와 같아요.

- 양적으로 차이가 나기에 이력서보다 많은 리소스가 들어감
- 단순 텍스트인 이력서와 달리 사진, 앨범형태UI 등 docx로 관리하기엔 제어하기 어려운 부분이 다수 존재
- 당장 포폴을 꾸미거나 정리하는데 시간이 부족하다면 notion을 이용해 시간을 아끼고 다른 곳에 안배하는 것이 적절

어디까지나 제 의견이지만, 
대다수의 경쟁자들이 notion을 사용한다면 본인만의 아이덴티티를 나타낼 수 있는 디자인을 가져가는 것도 하나의 전략이 될 수 있을 것 같아요.
그러나 시간이 많이 소요되고 여기에 가중치를 두다 보면 오히려 중요한 부분(면접 준비 등)을 놓칠 수 있기에 그닥 우선순위는 높지 않다고 정리할 수 있겠네요.

제 답변은 참고만 하되 @tneserp 님의 주관대로 생각을 잘 해보시고 결정하시길 바랍니다!
취업 잘 준비하시기 바랍니다 :)

**Reply by tneserp** [reply]
감사합니다. 저도 SSAFY 10기로서, 현재 특화 프로젝트를 진행하고 있습니다. 자율 프로젝트를 앞두고 기획에 관해 많은 고민이 있는데요.

백엔드로서 Spring, Spring Security, MySQl, JPA, Redis와 같이 진부하다면 진부한 기술들만 썼는데요. 그래서 자율 프로젝트에서는 취업시장에서 신입으로서 어필할만한 기술들을 가져가기 위해 팀원들 모두가 백엔드인 팀을 결성했습니다. (상을 포기했습니다.) (https://github.com/wlwlsus/can-you-get-it?tab=readme-ov-file 와 같은 프로젝트를 지향하고 있다고 생각하시면 될 것 같습니다.)

위의 답글을 보고 what 보다는 how, why가 중요하다는 것은 알지만 기획 단계에 있고, 각자 가져가고 싶은 기술을 사용할 수 있는 프로젝트를 하자고 해서 고민입니다.  그래서 어떤 기술을 가져가야 하나 고민중인데 .. 해주실 말씀이 있으실까요?

**Reply by CODe5753** [reply]
@tneserp 
굉장히 바쁜 나날을 보내고 계시겠군요~
아직 구체적인 도메인이 정해진 것이 아니어서 기술적인 부분을 조언드리긴 어렵네요.
다만, 어떤 아이디어를 정하실 때 "설명하기 쉬운" 비즈니스 도메인을 선택하시기 바랍니다.
면접에서도 중요한 부분인데요, **내가 진행했던 프로젝트를 면접관이 이해하지 못하면 거기서 소모되는 시간이 꽤 큽니다!**

즉, 일상생활에서 접하기 쉽고 특별히 비즈니스를 설명하지 않아도 이해가 될 만한 아이디어를 선정하시기 바랍니다!

**Reply by tneserp** [reply]
감사합니다! 구체적인 도메인이 정해지진 않았지만 대용량 트래픽에 관한 주제를 진행할 것 같은데요!

위에 기재해놓은 can-you-get-it 과 비슷하게 신발 드로우 같은 거, 수강신청 등을 주제로 한다면 어떤 기술을 써보는게 좋을지 말씀해주실 수 있을까요?

너무나 많은 기술이 있고, 다 할 수 있다면 좋겠지만 여건이 되지 않고, 1~2가지의 기술을 깊게 적용하는 것도 중요하겠지만, `왜? 적용했는가`를 중요시 여기면서 학습하고 싶어서요.!

**Reply by CODe5753** [reply]
@tneserp 

> can-you-get-it 과 비슷하게 신발 드로우 같은 거, 수강신청 등

아마 **예매, 재고관리 등의 동시성 및 정합성 이슈가 발생할 수 있는 상황**을 말씀하시는 것 같아요.

수강신청을 예로 들어볼까요.
이미 잘 아시겠지만, 평소엔 딱히 트래픽이 없다가 특정 시간에 도달하면 순간적으로 트래픽이 치솟는 시스템이죠. 마치 **선착순 쿠폰 발급 시스템**처럼요.
`대용량 트래픽`에 초점을 맞춰 조금 넓은 영역에서 얘기해보자면 `분산 시스템 설계`, `로드 밸런싱`, `캐싱`, `데이터베이스 파티셔닝/샤딩`, `비동기 처리`, `메시지 큐` 등이 있겠네요.
비즈니스적으로 이슈가 없으려면 데이터 정합성에 대해서도 고민이 필요하겠고, 임계 영역에 접근 가능한 요청에 제한이 있을텐데 **무조건적인 확장 설계를 하는 것이 과연 필수적인지** 고민도 해봐야겠네요. 
동시에 어떤 락 방식을 사용할지 고민하는 부분도 재밌겠네요.

더 구체적으로 답변 드리고 싶지만 지금 질문주신 내용에서 제가 조언드릴 수 있는 부분은 여기까지인 것 같아요.

제가 조언 드릴 때의 접근법은 **프로젝트의 명확한 목표와 스코프를 정의**하는 것에서 시작되는데, 이는 기술 선택에 있어서도 중요한 기준이 되거든요.
따라서 조언을 드리기엔 조금 추상적인 질문이라 **답변할 수 있는 범위가 넓어 말씀드리기 어려운 상황**이에요.

**주제를 선정**한 후 극한의 상황에서 **어떤 기술로 해결할 수 있는지 고민**해보시고
그 안에서 여러 선택지가 있을텐데 **선택지들끼리 비교해본 후** 지금 현실적 여건(데드라인, 구현 난이도, 비즈니스)을 살펴보세요.
그러면 훨씬 더 구체적인 질문이 만들어질 것 같아요!

+)
도움이 되지 못한 것 같아 아쉽네요 T-T
궁금한 점이 있으시다면 언제든 질문하러 오세요~!

**Reply by tneserp** [reply]
감사합니다! 비동기 처리, 메시지 큐, 로드밸런싱 등을 공부해보고 프로젝트에 녹여낼 수 있는지 사용해봐야겠어요! 감사합니다!

**Comment by minjaeleedev** [comment]
덕분에 이력서 작성에 정말 많은 참고가 되었습니다. 감사합니다.

**Reply by CODe5753** [reply]
참고가 되셨다니 다행이네요. 감사합니다 ☺️

**Comment by mag123c** [comment]
안녕하세요 승현님!!!
혹시 최종 형태로 작성한다고 하면 경력기술서 같은 프로젝트 상세 내용(?)은 따로 제출하시나요?

**Comment by bibisam06** [comment]
승현님 신입개발자로 취업준비중인데 너무 도움 되었습니다! 저도 승현님같은개발자되려구노력해야겟습니다...
